### ISSUE TITLE
Question: Would it be worth while to consolidate documentation

### ISSUE THREAD

**Issue by nutterb** [vubiostat/redcapAPI#216]
For example, right now we have `deleteArms`, `exportArms`, and `importArms`, which all open different help pages. Would it be of interest to consolidate these all so that they all share the same help file?

If so, do we also want to consolidate the code files so that they all exist in the same file? I can make a case for this either way. I generally like having each function be in a separate file, but I also like functions to be in the same file of their documentation.

If not, we should at least utilize the `@seealso` tag in the documentation to link the different methods to each other.

Groupings to consider are:

1. `deleteArms`, `exportArms`, `importArms`
2. `deleteDags`, `exportDags`, `importDags`
    + optionally: `switchDag`, `exportUserDagAssignments`, `importUserDagAssignments`
3. `deleteEvents`, `exportEvents`, `importEvents`
4. `exportMetaData`, `importMetaData`, `exportFieldNames`
5. `deleteFile`, `exportFile`, `importFile`
6. `deleteFileFromRepository`, `exportFileFromRepository`, `importFileToRepository`
    + optionally: `exportFileRepositoryListing`, `createFileRepositoryFolder`
7. `deleteFileRepository`, `exportFileRepository`, `importFileRepository`
8. `exportInstruments`, `exportMappings`, `importMappings`, `exportPDF`
9. `exportLogging`
10. `exportProjectInformation`, `importProjectInformation`
11. `deleteRecords`, `exportRecords`, `importRecords`, `exportNextRecordName`, `renameRecord`, `exportReports`
12. `exportRecordsTyped`
13. `exportRepeatingInstruments`, `importRepeatingInstruments`
14. `deleteUsers`, `exportUsers`, `importUsers`
15. `deleteUserRoles`, `exportUserRoles`, `importUserRoles`
    + Optionally: `exportUserRoleAssignments`, `importUserRoleAssignments`

**Comment by nutterb** [comment]
The following is an approximation of what the consolidated documentation might look like. 


## Export, Import, Delete Data Access Groups from a Project

### Description
These methods allow you to export existing Data Access Groups, import new Data Access Groups, or delete Data Access Groups from a project.

### Usage
```
deleteDags(rcon, dags, ...)

## S3 method for class 'redcapApiConnection'
deleteDags(
  rcon,
  dags,
  refresh = TRUE,
  ...,
  error_handling = getOption("redcap_error_handling"),
  config = list(),
  api_param = list()
)

exportDags(rcon, ...)

## S3 method for class 'redcapApiConnection'
exportDags(
  rcon,
  ...,
  error_handling = getOption("redcap_error_handling"),
  config = list(),
  api_param = list()
)

importDags(rcon, data, ...)

## S3 method for class 'redcapApiConnection'
importDags(
  rcon,
  data,
  refresh = TRUE,
  ...,
  error_handling = getOption("redcap_error_handling"),
  config = list(),
  api_param = list()
)
```
### Arguments
* `rcon` 	A `redcapConnection` object.
* `dags`      `character` vector of names matching the `unique_group_name`.
* `...`	Additional arguments to pass to other methods.
* `refresh`	`logical(1)`. When `TRUE`, cached event data will be refreshed after the import.
* `error_handling`	An option for how to handle errors returned by the API. see `redcapError`
* `config` `list` Additional configuration parameters to pass to `POST`. These are appended to any parameters in `rcon$config`.
* `api_param`	`list` Additional API parameters to pass into the body of the API call. This provides users to execute calls with options that may not otherwise be supported by redcapAPI.
* `data`	A `data.frame` with two columns: `data_access_group_name` and `unique_group_name`.

### Details
When importing data access groups, provide a value for `data_access_group_name` with no value for `unique_group_name`.

To modify a group name, provide a new value for `data_access_group_name` with the associated `unique_group_name`. If `unique_group_name` is provided, it must match a value currently in the project.

### Value
`exportDags` returns a data frame with the columns
* `data_access_group_name` The human readable name for the data access group.
* `unique_group_name` The internal unique group name
* `data_access_group_id` The internal numeric identifier.

`importDags` has no return, but will print a message indicating the number of Data Access Groups imported.

`deleteDags` has no return, but will print a message indicating the number of Data Access Groups deleted.

### See Also
`exportUserDagAssignments`, `importUserDagAssignments`

**Comment by spgarbet** [comment]
I think this makes a lot of sense.

**Comment by nutterb** [comment]
I'm not going to be reckless enough to do all of this in one pull request.  I'll pick away at these when I've got spare time

- [ ] deleteArms, exportArms, importArms
- [ ] deleteDags, exportDags, importDags
    + See also: switchDag, exportUserDagAssignments, importUserDagAssignments
- [ ] switchDag
    + See also: deleteDags, exportDags, importDags, exportUserDagAssignments, importUserDagAssignments
- [ ] exportUserDagAssignments, importUserDagAssignments
    + deleteDags, exportDags, importDags, switchDag
- [ ] deleteEvents, exportEvents, importEvents
- [ ] exportMetaData, importMetaData, exportFieldNames
- [ ] deleteFile, exportFile, importFile
- [ ] deleteFileFromRepository, exportFileFromRepository, importFileToRepository
    + See also: exportFileRepositoryListing, createFileRepositoryFolder, deleteFileRepository, exportFileRepository, importFileRepository
- [ ] deleteFileRepository, exportFileRepository, importFileRepository
    + See also: exportFileRepositoryListing, createFileRepositoryFolder, deleteFileFromRepository, exportFileFromRepository, importFileToRepository
- [ ] exportFileRepositoryListing, createFileRepositoryFolder
    + See also: deleteFileRepository, exportFileRepository, importFileRepository, deleteFileFromRepository, exportFileFromRepository, importFileToRepository
- [ ] exportInstruments, exportMappings, importMappings, exportPDF
- [ ] exportLogging
- [ ] exportProjectInformation, importProjectInformation
- [ ] deleteRecords, exportRecords, importRecords, exportReports
    + See also: exportNextRecordName, renameRecord, exportRecordsTyped, exportReportsTyped
- [ ] exportNextRecordName, renameRecord, 
    + See also: deleteRecords, exportRecords, importRecords, exportReports, exportRecordsTyped, exportReportsTyped
- [ ] exportRecordsTyped, exportReportsTyped
    + deleteRecords, exportRecords, importRecords, exportReports, exportNextRecordName, renameRecord
- [ ] exportRepeatingInstruments, importRepeatingInstruments
- [ ] deleteUsers, exportUsers, importUsers
    + See also: deleteUserRoles, exportUserRoles, importUserRoles, exportUserRoleAssignments, importUserRoleAssignments
- [ ] deleteUserRoles, exportUserRoles, importUserRoles
    + See also: deleteUsers, exportUsers, importUsers, exportUserRoleAssignments, importUserRoleAssignments
- [ ] exportUserRoleAssignments, importUserRoleAssignments
    + See also: deleteUsers, exportUsers, importUsers, deleteUserRoles, exportUserRoles, importUserRoles


General template / style guide

1. Use third person throughout
2. Use "the user" instead of "you"
3. remove all `@author` tags
4. When encountering `@source`, confer with others about whether it should be included in the file level, or the package level

```
@name [system]Methods # example armsMethods
@alias methodName
@title .....
@description .....

@param [parameter name] \code{R type} description of the argument

@details ....

@return Describe the return for each function. When data frames are returned, return itemized list of expected columns.
```

**Comment by nutterb** [comment]
I think this is ready for an independent set of eyes before I dig in much further. 

I think I'm satisfied with the appearance. Still struggling a bit with organization though.  I've been putting all of the documentation in a file called `documentation.R`, but that file is turning into a huge wall of text, which is a bit overwhelming. I can't settle on whether I prefer (using the Arms methods as an example)

1. Keep the documentation separate, with distinct files for exportArms, importArms, and deleteArms
2. Put the documentation in the deleteArms.R file, but keep each function in its own file. 
3. Put the documentation and the functions all in the same file

Modified documentation can be viewed in https://github.com/nutterb/redcapAPI/tree/Issue-216-consolidateDocumentation

**Comment by spgarbet** [comment]
I like the idea of functions in separate files when possible. I think it's easier to quickly find them. Conversely the find in files function is what I always end up using, so I would say this preference is more idealistic than real.

I think there's 4th option, keep the documentation for each group in it's own file. I.e., have a file `docArms.R`, `docDags.R`. This would probably be my approach. 

I skimmed over your examples and it's looking really good. I've cleaned up corners here and there, but this pulls it together cohesively. I have someone on staff who has automated grammar tools that can do full passes over it when you're ready. I' can do an editing pass as well.

**Comment by nutterb** [comment]
I think I'm ready to have it passed through the grammar tools. 

A couple things you should be aware of

1. I made a few documentation objects for the sole purpose of using `@inheritParams`. This ensures that the documentation of arguments like `config` and `api_param` are consistent throughout the packages. (see `documentation-common-args.R`
2. I was fighting with roxygen to get some content in the order I wanted it and found it was easier to get what I wanted if I switched on the markdown parsing (see `DESCRIPTION`).  A side benefit of this was that making tables and lists was much easier. So I went ahead and converted all the `\code{}` and `\code{\link{}}` tags to their markdown equivalents.
3. I didn't spend a lot of time making examples for unexported functions. But I did make sure third person was in use and arguments were documented consistently. 
4. I followed my grouping above fairly closely, but found that `importRecords`, `exportRecords`, and `deleteRecords` were dissimilar enough that they didn't fit well together. I also decided `importRecords` is an important enough topic that it needs to be addressed on its own (its help file isn't long, but it is focused).
5. All references to external sources have been removed (they were all either "refer to your institution's documentation" or "refer to the package wiki" which is pretty defunct)

I'm sure I've missed quite a few things.  Every time I look at a file I find something else that could be a little bit better.  Hypercriticism is welcome.

https://github.com/nutterb/redcapAPI/tree/Issue-216-consolidateDocumentation

**Comment by spgarbet** [comment]
Request forwarded to QA.

**Comment by sharpwb** [comment]
here are a few spelling grammar things that I found just now

1- check this "[field_labe]" where [field_label] -- not sure this is what was desired
2- identified as checkbox variables are relabelled --> should be relabeled
3- logical values and are also not casesensitive  --> case sensitive
4- can not should be cannot --- search and replace for a few instances of this
5-The cast override is one were users can specify  – were should be where
6- An error is returned if an value in the data — "an" should be "a"
7- column is not provided, user will not assigned to any group --> should be "column is not provided, user will not **be** assigned to any group
8-f the ’unique_role_name’ column is not provided, user will not assigned to any user role --> not assigned should be "not be assigned"
9-If TRUE, the all caches in the connection object will be flushed ---> I think this is supposed to be "If TRUE,  then all caches in the connection object will be flushed"
OR "If TRUE, all of the caches in the connection object will be flushed"

**Comment by sharpwb** [comment]
I used a 'Consistency" tool that suggests these , Once again these are just suggestions and may vary with audience

the following were found  by a consistency checker

‘event form’ / ‘event-form’1 time without a hyphen2 times with
‘instrument event’ / ‘instrument-event’1 time without a hyphen3 times with
‘post processing’ / ‘post-processing’1 time without a hyphen3 times with
Randomization-Module’ / ‘Randomization Module’1 time with a hyphen2 times without
‘read-only’ / ‘read only’3 times with a hyphen3 times without
‘User-DAG’ / ‘User DAG’1 time with a hyphen1 time without
‘User-Data’ / ‘User Data’1 time with a hyphen4 times without
‘User-Role’ / ‘User Role’3 times with a hyphen11 times without
‘user-specified’ / ‘user specified’1 time with a hyphen9 times without
‘casesensitive’ / ‘case-sensitive’1 time without a hyphen2 times with
DeIdentified’ / ‘De-Identified’2 times without a hyphen13 times with
‘email’ / ‘e-mail’1 time without a hyphen3 times with
‘sub-folder’ / ‘subfolder’1 time with a hyphen1 time without
‘sub-folders’ / ‘subfolders’2 times with a hyphen5 times without
‘backward’ / ‘backwards’backward 5 timesbackwards 1 time
‘labelled’ / ‘labeled’labelled 17 timeslabeled 1 time
Numbers 1 to 10spelled out 30 times (e.g. ‘wo randomization’, ‘of two ’, ‘least two arms’) but numerals 11 times (e.g. ‘with 0 rows’, ‘of 0 actually’, ‘have 0 as’)


Other suggestions from a misc tool

CONTRACTIONS
If you are writing a formal document then consider spelling out these contractions in full:

 
can’t (1 time)
doesn’t (13 times)
don’t (4 times)
hasn’t (1 time)
haven’t (1 time)
isn’t (5 times)
weren’t (1 time)

**Comment by nutterb** [comment]
Thanks @sharpwb , that was extremely helpful.

* check this "[field_labe]" where [field_label] -- not sure this is what was desired **corrected**
* identified as checkbox variables are relabelled --> should be relabeled **all non-code instances now use relabeled (see details below**
* logical values and are also not casesensitive --> case sensitive **corrected**
* can not should be cannot --- search and replace for a few instances of this **corrected**
* The cast override is one were users can specify – were should be where **corrected**
* An error is returned if an value in the data — "an" should be "a" **corrected**
* column is not provided, user will not assigned to any group --> should be "column is not provided, user will not be assigned to any group **corrected**
* f the ’unique_role_name’ column is not provided, user will not assigned to any user role --> not assigned should be "not be assigned" **corrected**
* If TRUE, the all caches in the connection object will be flushed ---> I think this is supposed to be "If TRUE, then all caches in the connection object will be flushed" OR "If TRUE, all of the caches in the connection object will be flushed" **corrected**
* ‘event form’ / ‘event-form’1 time without a hyphen2 times with **The one instance of 'event form' without the hypen is in the context of 'the adverse event form'. The other instances are in context of event-form mappings.  No changes made**
* ‘instrument event’ / ‘instrument-event’1 time without a hyphen3 times with **all instances now use instrument-event**
* ‘post processing’ / ‘post-processing’1 time without a hyphen3 times with **all instances now use post-processing**
* Randomization-Module’ / ‘Randomization Module’1 time with a hyphen2 times without **all instances now use randomization module**
* ‘read-only’ / ‘read only’3 times with a hyphen3 times without  **all instances are consistent with REDCap documentation. See details below. No changes made.**
* ‘User-DAG’ / ‘User DAG’1 time with a hyphen1 time without **all instances now use User-Dag**
* ‘User-Data’ / ‘User Data’1 time with a hyphen4 times without **all instances now use User Data**
* ‘User-Role’ / ‘User Role’3 times with a hyphen11 times without **updated so that all appearances of User Role refer to the object of a user role, and all appearances of User-Role refer to the mapping a user to a user role object.**
* ‘user-specified’ / ‘user specified’1 time with a hyphen9 times without **all instances are now 'user specified'**
* ‘casesensitive’ / ‘case-sensitive’1 time without a hyphen2 times with **all instances are now 'case-sensitive'**
* DeIdentified’ / ‘De-Identified’2 times without a hyphen13 times with **all instances are now de-identified**
* ‘email’ / ‘e-mail’1 time without a hyphen3 times with **in code all instances are 'email'. In documentation, all instances are now 'e-mail'**
* ‘sub-folder’ / ‘subfolder’1 time with a hyphen1 time without **all instances are now 'subfolder'**
* ‘sub-folders’ / ‘subfolders’2 times with a hyphen5 times without **all instances are now 'subfolders'**
* ‘backward’ / ‘backwards’backward 5 timesbackwards 1 time **all instances are now 'backward'**
* ‘labelled’ / ‘labeled’labelled 17 timeslabeled 1 time **addressed previously. See details below.**
* Numbers 1 to 10spelled out 30 times (e.g. ‘wo randomization’, ‘of two ’, ‘least two arms’) but numerals 11 times (e.g. ‘with 0 rows’, ‘of 0 actually’, ‘have 0 as’) **all remaining instances of 0-10 as numerals have been reviewed. Those that remain are referring to codings from REDCap or things like 'return a data frame of 0 rows', where I prefer the numeral, since it looks like what is returned by `nrow`**
* Removed contractions by reviewing all instances of `'t `. I also removed contractions from warnings issued in the code; I'll likely need to do some work on tests as a result.

When discussing labels in the documentation, all instances have been changed to "labeled" or "labeling." In code, however, I have continued to use "labelled," because this is how the attribute is defined in the Hmisc package. 

For read only vs read-only, the API documentation uses both. When user permissions to access forms, the value 2 is mapped to "Read Only" and the value 1 is mapped to "View records/responses and edit records (survey responses are read-only)."

**Comment by nutterb** [comment]
Closed via #234
